### DISCUSSION TITLE
sensor.streaming_radio_stations

### DISCUSSION THREAD

**Discussion by oleggogol** [jazzyisj/home-assistant-config#41] [General]
Is the sensor.streaming_radio_stations creation code publicly available?
If so, where can I view it?

**Comment by jazzyisj** [comment]
I never made it public.  It was something I was fooling around with and never really finished.  All it really did was take a manually created CVS file of streaming radio station URLs and created a sensor that listed those URLS as an attribute so I could reference them in HA.

If there is a particular bit of code you are looking for let me know, I can post it here for you.

I've recently switched my entire "media" config over to Music Assistant which is a major change to my config that **greatly** simplifies everything.  I'll be uploading all those changes once I'm done debugging the changeover in my config.

https://github.com/music-assistant/hass-music-assistant

**Reply by oleggogol** [reply]
Thanks for the quick reply,
I apologize for my English - I am writing through Google translator.
I have read all the forum posts on this issue, including yours.
I cannot transfer the CVS file to sensor.streaming_radio_stations (the sensor is not created)


my files:

python_scripts/internet_radio.py
------------------------------------------
import csv

csv_file = '/config/www/internet_radio.csv'

csv_reader = csv.DictReader(open(csv_file), delimiter=';')
station_list = [dict(d) for d in csv_reader]

sensor = 'streaming_radio_stations'
state = 'ok'
attributes = '{"stations": ' + str(station_list) + '}'

data = '{"state": "' + str(state) + '", "attributes": ' + attributes + '}'
data = data.replace("'",'"')
print(data)
----------------------------------------------

launching python_scripts

automation:

    - alias: "Radio - script on"
      id: radio_script_on
      trigger:
        - platform: homeassistant
          event: start
      action:
        - service: python_script.internet_radio
          data: {}

Maybe you can tell me where the error is !!!

**Reply by jazzyisj** [reply]
Is this how you are running your python script?  https://home-assistant.io/integrations/python_script/

If it is you I think you missed this note.
![image](https://user-images.githubusercontent.com/44064559/178751785-762496e3-4266-4c7e-b048-2b32bcc62977.png)

Also I believe you'll have to create your sensor using the command line sensor to run the python script.  I don't think you can create a sensor with just a service call.

https://www.home-assistant.io/integrations/sensor.command_line/

I did not create my sensor with a python script, I created my own custom component that creates the sensor.

If you want to try that, I would use this to get started.  https://github.com/custom-components/integration_blueprint

But honestly that is way overkill.   You could just make a template sensor if you don't need to keep your list in a CVS file.  I had a very specific reason for doing that.

```
template:
  - sensor:
      - name: "Playlists Sensor"
        unique_id: playlists_sensor
        icon: mdi:spotify
        state: ok
        attributes:
          playlists: >
            {{
              [
                {
                  "name":"Funk Outta Here",
                  "uri":"spotify:playlist:37i9dQZF1DX70TzPK5buVf"
                },
                {
                  "name":"70s Road Trip",
                  "uri":"spotify:playlist:37i9dQZF1DWWiDhnQ2IIru"
                },
                {
                  "name":"2000's Hip Hop",
                  "uri":"spotify:playlist:01pNIDYGqmeawppy32wr3D"
                }
              ]
            }} 
```

**Reply by oleggogol** [reply]
Thanks, I'll try to solve it
